### DISCUSSION TITLE
Is it possible to map from multiple sources?

### DISCUSSION THREAD

**Discussion by wbuck** [riok/mapperly#680] [Q&A]
Is it possible to map from multiple source objects?

```
record Target
{
    public required string Name { get; init; }
    public required string Phone { get; init; }
}

class Source1
{
    public string Name { get; set; }
}

class Source2
{
    public string Phone { get; set; }
}
```

```
public static partial Target Map(Source1 s1, Source2 s2);
```

I presume no as I'm not sure how that would work.

**Comment by TimothyMakkison** [comment]
Hey, thanks for your question. While Mapperly doesn't explicitly support multiple source mapping, it is possible to implement this.

Instead you could try chaining together two [existing target mappers](https://mapperly.riok.app/docs/configuration/existing-target/) like so:

```C#
[Mapper]
public static partial class Mapper
{
    private static partial Target MapSource1(Source1 src);

    private static partial void MapSource2(Source2 src, Target trg);

    public static Target Combine(Source1 src1, Source2 src2)
    {
        var target = MapSource1(src1);
        MapSource2(src2, target);
        return target;
    }
}
```

It's a bit verbose and won't work with `init`/`required` properties but it should work.

Hope this helps 😄 

When #622 or #587 are added rhis will be a lot easier.
